resources/scponly_user.rb

```diff
@@ -0,0 +1,126 @@
+# true To learn more about Custom Resources, see https://docs.chef.io/custom_resources.html
+
+resource_name :scponly_user
+
+default_action :create
+
+property :write_dir, String, default: 'incoming'
+property :public_key, String
+property :private_key, String
+property :chroot, [true, false], default: true
+property :altroot, String
+property :binaries,
+          Array,
+          default:
+          %w(
+            /bin/chgrp
+            /bin/chmod
+            /bin/chown
+            /bin/ln
+            /bin/ls
+            /bin/mkdir
+            /bin/mv
+            /bin/rm
+            /bin/rmdir
+            /usr/bin/scp
+            /usr/sbin/scponlyc
+          )
+
+action :create do
+  run_context.include_recipe 'osl-scponly::default'
+
+  if new_resource.chroot
+
+    altroot = new_resource.altroot.nil? ? '/home/chroot' : new_resource.altroot
+
+    directory "#{altroot}/home/#{new_resource.name}" do
+      recursive true
+    end
+
+    user new_resource.name do
+      gid 'scponly'
+      manage_home true
+      home "#{altroot}/home/#{new_resource.name}"
```

Comment:
So I was wrong about this and saw this mentioned in the ``BUILDING-JAILS.TXT`` file:

```
   The two slashes ('//') are used by scponlyc to determine the chroot
   point.  Everything in front of them is used as the new root location
   in the chroot call.  Everything after the double slash designates the 
   directory to chdir() into AFTER chrooting.  This is so users can be
   dropped into a writable directory inside the chroot.
```


```suggestion
      home "#{altroot}//home/#{new_resource.name}"
```